### PR TITLE
ARXIVNG-2315 user should not be able to skip required steps

### LINKED ISSUE #113: It should not be possible to skip the verify user step
Problem: On a new submission, I can skip the verify user step and go straight to 'upload files'.

To reproduce: From "Manage Submissions" click "Start a New Submission." Then click the "Upload files" on the stages progress bar.

Behavior: Clicking "Upload Files" brings up the upload files stage form.
Expected Behavior: A warning message "Please verify your personal information before proceeding." and staying on the Verify User form.

### PR DESCRIPTION
**Pull request by erickpeirson**
This fixes #113 . There was a logical error in the method that determines whether or not a step can be visited. Added some extra assertions to app test to verify.

### CODE REVIEW

File: submit/util.py [hunk 1]
```diff
@@ -7,13 +7,14 @@
...
 from arxiv.base import logging
 from arxiv.base.globals import get_application_global
+from arxiv.submission.services.classic.exceptions import Unavailable
 import arxiv.submission as events
 
 logger = logging.getLogger(__name__)
 logger.propagate = False
 
 
-@retry(tries=3, delay=0.25, backoff=3)
```
Comment (bdc34) [inline submit/util.py:16]: The changes to this line don't seem related to this bug. Do they belong in a different PR or are the trivial or are they part of some debugging work?

**Review by bdc34 — APPROVED** [review]

File: submit/util.py (hunk 1 above)
Comment (erickpeirson) [reply submit/util.py:16]: Yeah, was just opportunistic tuning
